docs/docs/guides/labs/components/creating-new-component-types/creating-dg-plugin.md

```diff
@@ -0,0 +1,91 @@
+---
+title: 'Creating a `dg` plugin'
+sidebar_position: 200
+---
+
+import Preview from '@site/docs/partials/\_Preview.md';
+
+<Preview />
+
+A `dg` plugin is a Python package that defines and exposes `dg`-legible
+objects. `dg`-legible objects are collectively called _plugin objects_ and
+include custom component types, scaffolding classes, and more. Plugin objects
+that are exposed by an installed `dg` plugin can be inspected and interacted
+with by `dg` commands.
+
+Any Python package can be made into a `dg` plugin by declaring an [entry
```

Comment:
To make a package a plugin you add a section to your `pyproject.toml` like this:

```toml
[project.entry-points]
"dagster_dg.plugin" = { my_package = "my_package"}
```